srtp-ekt-diet/draft-ietf-perc-srtp-ekt-diet.md

```diff
@@ -274,16 +275,25 @@ this field is 32 bits.
 
 Security Parameter Index (SPI): This field indicates the appropriate
 EKTKey and other parameters for the receiver to use when processing
-the packet. The length of this field is 16 bits. The parameters
-identified by this field are:
+the packet, within a given conference. The length of this field is
+16 bits, representing a two-byte integer in network byte order. The
+parameters identified by this field are:
 
 * The EKT cipher used to process the packet.
 
 * The EKTKey used to process the packet.
 
 * The SRTP Master Salt associated with any master key encrypted with
   this EKT Key. The master salt is communicated separately, via
-  signaling, typically along with the EKTKey.
+  signaling, typically along with the EKTKey. (Recall that the SRTP
+  master salt is used in the formation of IVs / nonces.)
+
+Epoch: This field indicates how many SRTP keys have been used for this
```

Comment:
For a large and long meeting, might this change more than 256 times?  I'm thinking especially of conferences with a few hundred people.  If we follow the policy of changing keys each time a participant comes and goes, we would definitely roll this over.  We could add a length indicator on the front or just encode it so that the MSb is an extension bit indicating another octet follows.  A 0 would mean the next 7 bits are the last.  A 1 would mean take the next 7 bits as the MSbs and add the bits from the next octet (following the same extension rule).  For example, epoch 0b10101101 0b10011111 0b01011111 would equate to numeric value 0b00001011 0b01001111 0b11011111 or 0x0B4FDF.  I doubt we'd see one that long, but that's just to illustrate my suggestion.  This would work if we know the length of EKT Cipher, which I assume we would.  If not, we need an explicit length indicator or a larger field field.

Reply:
Note that this is a 16-bit field, so you get 65536 changes.  I am **not** going to do a varint.  Either we go `uint32`, or we leave it, or we drop it.

Reply:
I worry about fixed limits on anything.  I suspect 64K might be enough changes, but what happens when it's not?  If this epoch was also tied to the SPI value, then that solves the problem. I didn't see that stated explicitly.  That is, each time the KEK changes (and SPI is updated), the epoch gets reset.

Reply:
That's a good idea.  I've updated the text to scope the epoch to within SPI changes.